openlibrary/templates/site/banner.html

```diff
@@ -0,0 +1,22 @@
+$def with (announcement, storage_key)
+
+$ hidden = False
+$if ctx.user and ctx.user.preferences().get('hidden-banner') == storage_key:
+    $ hidden = True
+
+$def hideBanner():
+    $ctx.user.save_preferences({'hidden-banner': storage_key})
+
+$# Paste next line into the span element to show the blue banner
+$# <strong>Title</strong> $:announcement &nbsp; <a href="#" data-ol-link-track="AnnouncementsBar|BlogWidget" class="btn primary">Learn More</a>
+$if not hidden:
+    <div data-storage-key="$storage_key" class="page-banner--dismissable">
+        <span class="page-banner--dismissable-content">$:announcement</span>
+        $if ctx.user:
+            <button class="page-banner--dismissable-close" onclick="
+                $hideBanner();
+                window.location.reload();
```

Comment:
Let's avoid inline CSS and JS.

I didn't want you to remove all of the JS that you wrote before; I only asked that the key not be stored in `localStorage`.  
If somebody clicks the "X", do the following:
1. Hide the banner
2. Send a request to the server that will update the patron's preferences.

You will have to create a new handler for step 2.  You can add this to [/openlibrary/plugins/openlibrary/api.py](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/openlibrary/api.py).